sympy/functions/combinatorial/numbers.py

```diff
@@ -1715,22 +1715,25 @@ def _stirling1(n, k):
         return S.One
     if 0 in (n, k):
         return S.Zero
-    n1 = n - 1
 
     # some special values
     if n == k:
         return S.One
-    elif k == 1:
-        return factorial(n1)
-    elif k == n1:
+    elif k == n - 1:
         return binomial(n, 2)
     elif k == n - 2:
         return (3*n - 1)*binomial(n, 3)/4
     elif k == n - 3:
         return binomial(n, 2)*binomial(n, 4)
 
-    # general recurrence
-    return n1*_stirling1(n1, k) + _stirling1(n1, k - 1)
+    row = [1]+[0 for _ in range(k)]
```

Comment:
I think that `[0]*k` could be used.

Reply:
for row declaration? Would it make a significant difference?